src/mflux/utils/image_compare.py

```
@@ -0,0 +1,138 @@
+    num_mismatched = np.count_nonzero(~closeness_array)
+    total_elements = image1_data.size
+    mismatch_ratio = num_mismatched / total_elements
+    if mismatch_ratio > mismatch_threshold:
+        diff = image1_data - image2_data
+        # Scale the difference so it's visible. A difference of 50 will become bright.
+        # We'll scale it so that a difference of 50 or more becomes pure white (255).
+        diff_visual = (np.abs(diff) / 50 * 255).clip(0, 255).astype(np.uint8)
+        cv2.imwrite(image2_path.with_stem(image1_path.stem + "_diff"), diff_visual)
+        raise AssertionError(
+            f"{error_message_prefix} Check {image1_path} vs {image2_path} :: their elements are {mismatch_ratio:.1%} different. Fails assertion for {mismatch_threshold=}"
+        )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)